build/yamls/patches/ipsec/ipsecContainer.yml

```diff
@@ -0,0 +1,33 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: antrea-agent
+spec:
+  template:
+    spec:
+      containers:
+        - name: antrea-ipsec
+          image: antrea
```

Comment:
@antoninbas: I found no good solution to add the imagePullStrategy to IfNotPresent for dev builds. Let me know if you have an idea. Also for image url, I have to change to setting directly in generate-manifest.sh to make it apply to the ipsec container.

Reply:
I gave this some thought and came up with a proposed solution: https://github.com/antoninbas/antrea/commit/84fe31ecdbe59e11078c13fbf3f8417a9669bbc6. Basically I got rid of the `dev` and `release` overlays and instead generate a chain of overlays dynamically (from the `generate-manifest.sh` script) as appropriate: base -> ipsec -> kind -> dev / release. If we need to introduce another "feature" in the future, we can just add an overlay in the chain. Unfortunately there is a bit of clunkiness because of a lack of proper support to update the image pull policy globally: https://github.com/kubernetes-sigs/kustomize/issues/1493. Hopefully that will be resolved in the future.

Feel free to use my commit in your PR if it works for you.

Reply:
Sounds good to me.
But would you create a separate PR to merge your changes (without IPSec part) first? I will need more time to test and add e2e tests to merge this PR.

Reply:
What you think if I use sed or whatever to generate a kustomize path with generate-manifest.sh that takes the original config file with the revisions for IPSec?

Reply:
Ok I can do that today